### ISSUE TITLE
forecasting time series with lstm

### ISSUE THREAD

**Issue by darzok** [fchollet/keras#1603]
Hello

I have an issue when I try to forecast a time series with LSTM. For example, when my modele is trained I want to put in its input 12 values and I want it give me the 13th value.

to do that I trained my model with X_train = (nb_sample,timestep,1) 
with timestep = 12.

for example : 
when nb_sample = 0, input = 1,2,3,4,...,12
nb_sample = 1, input = 2,3,4,...,13

my modele seems to fit well and I get good score with my X_test sample, but when I want to test it, I give it 12 values by 12 values (exactly the same value that there are in my X_test sample) and it gave me a relly bad score. 

I think I forget something like statefull or mask but if it that, I don't really understand how it's work.
Please, it could be very helpful if someone can explain to me what is wrong with my code (I work with the lastest version of keras) :

```
model = Sequential()
model.add(LSTM(output_dim=60, input_dim=1, return_sequences=False))
model.add(Dense(input_dim=60, output_dim=1))
model.add(Activation("sigmoid"))

model.compile(loss='mse', optimizer=RMSprop(lr=0.0001, rho=0.9, epsilon=1e-6),class_mode="binary")

# let's load the weights from a save file.
model.load_weights('lstm_binary_8020_weight_645epoch.hdf5')

history = model.fit(X_train_2, y_train_bin, batch_size=128, nb_epoch=nb_epoch, show_accuracy=True, verbose=2, validation_data=(X_val_2, y_val_bin))
score = model.evaluate(X_test_2, y_test_bin, show_accuracy=True, verbose=2)

#save the weights
#model.save_weights('lstm_binary_sequence_by_sequence_weight_600epoch.hdf5',overwrite=True)`

```

**Comment by dvector89** [comment]
The model's final layer is a sigmoid function, whose output is in range of [0.0, 1.0]. Are you sure your target values are between 0.0 and 1.0?

**Comment by RavisMsk** [comment]
And sigmoid function is not the best choice for regressions problem. Just take linear or relu.
